lib/proto/handle-response.js

```diff
@@ -2,13 +2,13 @@ var clone = require("../util/clone.js")
 var newUid = require("../uniqueid.js")
 var trigger = require("../events/trigger.js")
 
-module.exports = function(responseText, request, href) {
-  var tempOptions = clone(this.options);
-  tempOptions.request = request
+module.exports = function(responseText, request, href, options) {
+  options = options || clone(this.options);
+  options.request = request
```

Comment:
I'm not sure which scenario you're trying to preempt.

Do you mean:
```
options = options ? clone(options) : clone(this.options)
options.request = request

//...

this.state.options = options
```

In case `handleResponse()` is called with `this.options`?

Reply:
What I'm trying to avoid is having `this.options` passed directly when events are triggered, as this PR changes `trigger(document, "pjax:error", tempOptions)` to `trigger(document, "pjax:error", options)`, for example. Passing the options object directly could allow `this.options` to be mutated accidentally. The snippet you posted would prevent that from happening.